Scripts/cours.js

```diff
@@ -56,47 +67,74 @@ function deleteAll(displayOnly = true){
 
     for(let i=size-1; i>1; i--){
         class_tab.deleteRow(i);
-    }
+    } 
+    showButton();
 }
 
 function deleteClass(id){
+    console.log(id);
+    if(classes == 0)
```

Comment:
classes is an array, checking if it's equal to 0 has no sense.
you want to check its length

Reply:
And why you are reinitializing the classes here ?

Reply:
What the difference between testing my array "classes" and its length?
If my array is empty, its lenght's is 0?

Reply:
I'm reinitilialize my array for recreate my array without class just deleted

Reply:
You have a problem here, when deleting an element all you have to do is to remove it from the classes variable. the test before is unnecessary

Reply:
And the difference is even if JS give us this possibility, but it's not a good practice. if you check the length of an array then show it so that your code will be easy to read